upgrade/upgrade.go

```diff
@@ -352,12 +353,18 @@ func waitForOldLonghornManagersToBeFullyRemoved(namespace, managerImage string,
 }
 
 func isOldManagerPod(pod corev1.Pod, managerImage string) (bool, string) {
+	runningPodVersion, exists := pod.GetLabels()["app.kubernetes.io/version"]
+
 	for _, container := range pod.Spec.Containers {
 		if container.Name == "longhorn-manager" {
-			if container.Image != managerImage {
+			// if the pod has a version label and the label does not match
+			// the image version, then the running pod (Manger) is old
+			// we are doing this since images can be mutated by webhooks
+			if exists && !strings.Contains(managerImage, runningPodVersion) || runningPodVersion == "" {
```

Comment:
Some thoughts

- This original condition check should be kept
   ```
      if container.Image != managerImage {
         return return true, container.Image
      } else {
          // Handle zarf case in this block
      }
```

- `managerImage` consists of image name and image tag. The tag might not have version string. Using `strings.Contains` is not a good solution.